File: TietoAcademyDemo.Domain/Book.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace TietoAcademyDemo.Domain
{
    public class Book
    {
        [Key]
        public string isbn { get; set; }

        public string title { get; set; }
        public DateTime? publishDate { get; set; }
        public string publishPlace { get; set; }
        public int numberOfPages { get; set; }
        public string subject { get; set; }
        public string coverUrl { get; set; }
        public string publisher { get; set; }
    }
}

File: TietoAcademyDemo/ClientApp/src/app/models/book.ts
export type BookId = string | number;

export interface Book {
  id?: BookId;
  authors: string;
  publisher: string;
  title: string;
  isbn: string;
  publishDate: Date;
  publishPlace: string;
  numberOfPages: string;
  subject: string;
  coverUrl?: string;
}
